Dockerfile

```diff
@@ -2,4 +2,24 @@ FROM openjdk:8-jre
 
 COPY ./modules/application/target/*SNAPSHOT.jar /truthtree/
 
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y git && \
+    curl -sL https://deb.nodesource.com/setup_8.x | bash - && \
+    apt-get install -y nodejs && \
+    npm install -g react-tools
+
+
+RUN git clone https://github.com/TruthTreeASD/cassiopeia.git
```

Comment:
Just curious, is there a reason you chose to compile front end inside the docker instead of inside travis ci's script?

The reason I'm asking because compiling front end code and put them into `./modules/application/src/main/resources/` before running `mvn clean install` at travis ci stage will create a self-contained `jar` file that can be run anywhere without relying on the `/public/` folder existence. 

Another benefit is that you don't have to install node and git for the docker container, and just use the plain old java base image.

Lmk what you think

Reply:
To compile front end code during Travis build, we will still need to install node in Travis script because we could not choose two languages. And it's hard to debug the installation error from Travis script which is also really time-consuming. If there will not be a big difference between using `/public/` and compacting the static files in the jar, I will choose to go with the current scripts.